### PR TITLE
Add AppEngine endpoints

### PR DESCRIPTION
**Pull request by szakhlypa**

### CODE REVIEW

File: lib/astarte/client/appengine/groups.ex [hunk 1]
```diff
@@ -19,6 +19,33 @@
...
+    with {:ok, %Tesla.Env{} = result} <- Tesla.get(tesla_client, request_path, query: query_opts) do
+      if result.status == 200 do
+        {:ok, result.body}
+      else
+        {:error, %APIError{status: result.status, response: result.body}}
+      end
+    end
+  end
+
+  def create(%AppEngine{} = client, data, group_name, devices \\ []) when is_binary(group_name) do
```
Comment (rbino) [inline lib/astarte/client/appengine/groups.ex:35]: As Github Actions is saying here below, `data` is unused here, remove it

File: lib/astarte/client/appengine/stats.ex [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Client.AppEngine.Stats do
+  alias Astarte.Client.{APIError, AppEngine}
+
+  def get(%AppEngine{} = client) do
```
Comment (rbino) [inline lib/astarte/client/appengine/stats.ex:22]: call it `get_devices_stats`, since we will probably introduce addidional stats in the future

**Review by rbino — CHANGES_REQUESTED** [review]

**Review by rbino — APPROVED** [review]

File: lib/astarte/client/appengine/devices.ex [hunk 3]
```diff
@@ -19,6 +19,19 @@
 defmodule Astarte.Client.AppEngine.Devices do
   alias Astarte.Client.{APIError, AppEngine}
 
+  def list(%AppEngine{} = client, query_opts \\ []) do
+    request_path = "devices"
+    tesla_client = client.http_client
+
+    with {:ok, %Tesla.Env{} = result} <- Tesla.get(tesla_client, request_path, query: query_opts) do
+      if result.status == 200 do
+        {:ok, result.body}
```
Comment (bettio) [inline lib/astarte/client/appengine/devices.ex:28]: so basically this is not the whole list of devices, but just a chunk of them. there will be anyway a next and a prev pointer, right?

Comment (bettio) [reply lib/astarte/client/appengine/devices.ex:28]: also: as soon as we get the first chunk, how do we get the next one? I can't understand how to provide it.

**Review by bettio — CHANGES_REQUESTED** [review]

File: lib/astarte/client/appengine/devices.ex (hunk 3 above)
Comment (szakhlypa) [reply lib/astarte/client/appengine/devices.ex]: Changed `list/2` to return whole list of devices in format {:ok, {"data" => []}} or an error

**Review by rbino — APPROVED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**